Source/WebKit/Shared/Cocoa/CoreIPCNSURLRequest.mm

```diff
@@ -0,0 +1,324 @@
+                },
+                [&] (const Vector<String> a) {
+                    NSMutableArray *mutArray = [NSMutableArray array];
+                    for (auto& item : a)
+                        [mutArray addObject: item];
+                    [mutDict setObject: mutArray forKey:(NSString *)headerPair.first];
+                }
+            );
+        }
+        [dict setObject: mutDict forKey:@"headerFields"];
+    }
+
+    SET_DICT_FROM_OPTIONAL_MEMBER(body);
+
+    if (m_data.bodyParts) {
+        NSMutableArray *mutArray = [NSMutableArray array];
+        for (auto& value : *m_data.bodyParts) {
+            WTF::switchOn(value,
+                [&] (const CoreIPCData d) {
+                    [mutArray addObject: d.toID().get()];
+                },
+                [&] (const CoreIPCString d) {
+                    [mutArray addObject: d.toID().get()];
+                }
+            );
+        }
+        [dict setObject: mutArray forKey:@"bodyParts"];
+    }
+
+    SET_DICT_FROM_PRIMITIVE(startTimeoutTime, NSInteger, Double);
+    SET_DICT_FROM_PRIMITIVE(requiresShortConnectionTimeout, NSInteger, Bool);
+    SET_DICT_FROM_PRIMITIVE(payloadTransmissionTimeout, NSInteger, Double);
+
+    [dict setObject:[NSNumber numberWithUnsignedChar: static_cast<uint8_t>(m_data.allowedProtocolTypes)] forKey:@"allowedProtocolTypes"];
+
+    SET_DICT_FROM_OPTIONAL_MEMBER(boundInterfaceIdentifier);
+    SET_DICT_FROM_OPTIONAL_PRIMITIVE(allowsExpensiveNetworkAccess, NSInteger, Bool);
+    SET_DICT_FROM_OPTIONAL_PRIMITIVE(allowsConstrainedNetworkAccess, NSInteger, Bool);
+    SET_DICT_FROM_OPTIONAL_PRIMITIVE(allowsUCA, NSInteger, Bool);
+    SET_DICT_FROM_PRIMITIVE(assumesHTTP3Capable, NSInteger, Bool);
+
+    [dict setObject:[NSNumber numberWithUnsignedChar: static_cast<uint8_t>(m_data.attribution)] forKey:@"attribution"];
+
+    SET_DICT_FROM_PRIMITIVE(knownTracker, NSInteger, Bool);
+    SET_DICT_FROM_OPTIONAL_MEMBER(trackerContext);
+    SET_DICT_FROM_PRIMITIVE(privacyProxyFailClosed, NSInteger, Bool);
+    SET_DICT_FROM_PRIMITIVE(privacyProxyFailClosedForUnreachableNonMainHosts, NSInteger, Bool);
+    SET_DICT_FROM_PRIMITIVE(privacyProxyFailClosedForUnreachableHosts, NSInteger, Bool);
+    SET_DICT_FROM_OPTIONAL_PRIMITIVE(requiresDNSSECValidation, NSInteger, Bool);
+    SET_DICT_FROM_PRIMITIVE(allowsPersistentDNS, NSInteger, Bool);
+    SET_DICT_FROM_PRIMITIVE(prohibitPrivacyProxy, NSInteger, Bool);
+    SET_DICT_FROM_PRIMITIVE(allowPrivateAccessTokensForThirdParty, NSInteger, Bool);
+    SET_DICT_FROM_PRIMITIVE(useEnhancedPrivacyMode, NSInteger, Bool);
+    SET_DICT_FROM_PRIMITIVE(blockTrackers, NSInteger, Bool);
+    SET_DICT_FROM_PRIMITIVE(failInsecureLoadWithHTTPSDNSRecord, NSInteger, Bool);
+    SET_DICT_FROM_PRIMITIVE(isWebSearchContent, NSInteger, Bool);
+
+    if (m_data.contentDispositionEncodingFallbackArray) {
+        NSMutableArray *mutArray = [NSMutableArray array];
```

Comment:
Either name it mutableArray or array.
Also, let's use initWithCapacity and adopt it to not fill up the autoreleasepool and to reduce allocations.  Here and a few other places in this file.